src/CoreShop/Bundle/OrderBundle/Controller/AbstractSaleDetailController.php

```diff
@@ -278,6 +279,10 @@ protected function getDetails(SaleInterface $sale)
             $jsonSale['priceRule'] = $rules;
         }
 
+        $event = new GenericEvent($sale, $jsonSale);
+
+        $this->get('event_dispatcher')->dispatch('coreshop.sale.detail.prepare', $event);
```

Comment:
- Adding `coreshop.sale.detail.prepare` as constant maybe?
- Passing `$jsonSale` as argument is not enough if guess (no reference). Adding a `$jsonSale = $event->getArguments()` after firing the event should do the job.

Reply:
done